src/selectors.ts

```diff
@@ -68,6 +68,23 @@ export const Components = {
         placeholder: '100',
       },
     },
+    Events: {
+      Query: {
+        label: 'Query',
+        tooltip: 'Sentry query to filter the results, details https://docs.sentry.io/product/sentry-basics/search/',
+        placeholder: 'is:unresolved',
+      },
+      Sort: {
+        label: 'Sort By',
+        tooltip: 'Sort results',
+        placeholder: 'Optional',
+      },
+      Limit: {
+        label: 'Limit',
```

Comment:
Is this the limit per page or the absolute limit? I think users will want to be able to graph this. Being able to see 100 rows of data may not be sufficient to quantify the impact of an incident. Consider this [graph](https://canva.sentry.io/discover/homepage/?field=title&field=event.type&field=project&field=user.display&field=timestamp&field=replayId&name=All+Events&project=1766513&query=&sort=-timestamp&statsPeriod=14d&yAxis=count%28%29). It makes 2 requests:

- one is to get the event data (i.e. the rows) (as is currently done via a call to the same API used here)
- another is to get the graph data (i.e. the x/y coordinates to generate a nice looking graph) (something like this `https://canva.sentry.io/api/0/organizations/canva/events-stats/?interval=30m&partial=1&project=1766513&query=&referrer=api.discover.default-chart&statsPeriod=14d&yAxis=count%28%29`)

I think both are required for this work - I think one of the most important parts of querying sentry by events is being able to see the number of certain events over time rather than the past 100.

Are we able to do that as part of a Grafana plugin? So, when the user executes a query, we return both the rows and a graphical representation of the data for the given timescale?

Reply:
1. This is a per-page limit. The response contains a header with a link for the next page. There is no pagination support in the current plugin implementation, but we can consider adding it later.

2. Regarding the provided [graph](https://canva.sentry.io/discover/homepage/?field=title&field=event.type&field=project&field=user.display&field=timestamp&field=replayId&name=All+Events&project=1766513&query=&sort=-timestamp&statsPeriod=14d&yAxis=count%28%29) example. 
You can achieve the desired result by using two panels - one for querying Stats and representing the number of events as a graph, and the second panel for the list of events itself.
Below is a screenshot where I was able to create a similar graph (the numbers are higher on the graph because it uses 1-hour frames).
![image](https://github.com/Canva/grafana-sentry-datasource/assets/20262077/efd4e7eb-95c6-4ec3-ad30-fcd75fca8517)